vllm/model_executor/layers/rejection_sampler.py

```diff
@@ -77,24 +82,33 @@ def forward(
         """
         # Only perform shape/dtype/device checking in strict mode, as it adds
         # overhead.
+        target_probs = target_with_bonus_probs[:, :-1]
         if self._strict_mode:
             self._raise_if_incorrect_input(target_probs, bonus_token_ids,
                                            draft_probs, draft_token_ids)
 
-        accepted, recovered_token_ids = (
-            self._batch_modified_rejection_sampling(
-                target_probs,
-                draft_probs,
+        if chain_speculative_sampling:
```

Comment:
nit: use `chain_speculative_sampling is not None` to be precise (None equality is different than [Falsey](https://stackoverflow.com/questions/39983695/what-is-truthy-and-falsy-how-is-it-different-from-true-and-false))

Reply:
suggest also adding a comment on this branch for readability